File: basics/two.js
const express = require('express');
const app = express();

app.use(express.json());

let items = [{ id: 1, name: "Item1" }];

app.get('/items', (req, res) => {
  res.json(items);
});

app.post('/items', (req, res) => {
  items.push(req.body);
  res.json(req.body);
});

app.listen(3000, () => console.log("Server running"));

File: basics/two.go
package main

import (
	"encoding/json"
	"net/http"
)

type Item struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
}

var items = []Item{
	{ID: 1, Name: "Item1"},
}

func getItems(w http.ResponseWriter, r *http.Request) {
	json.NewEncoder(w).Encode(items)
}

func addItem(w http.ResponseWriter, r *http.Request) {
	var newItem Item
	json.NewDecoder(r.Body).Decode(&newItem)
	items = append(items, newItem)
	json.NewEncoder(w).Encode(newItem)
}

func main() {
	http.HandleFunc("/items", getItems)
	http.HandleFunc("/add", addItem)
	http.ListenAndServe(":8080", nil)
}